lib/src/model/comment_referable.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+///
+/// Code for managing comment reference lookups in dartdoc.
+///
+library dartdoc.src.model.comment_reference;
+
+import 'dart:core';
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:dartdoc/dartdoc.dart';
+import 'package:meta/meta.dart';
+
+class ReferenceChildrenLookup {
+  final String lookup;
+  final List<String> remaining;
+  ReferenceChildrenLookup(this.lookup, this.remaining);
+}
+
+/// Support comment reference lookups on a Nameable object.
+mixin CommentReferable implements Nameable {
+  /// Look up a comment reference by its component parts.  If [parent] is true,
+  /// try looking up the same reference in any parents of [this].
+  @nonVirtual
+  CommentReferable referenceBy(List<String> reference, [bool parent = true]) {
```

Comment:
I think `parent` should probably be named. https://dart.dev/guides/language/effective-dart/design#avoid-positional-boolean-parameters